### PR TITLE
Allow information popovers while showing autocomplete suggestions

### LINKED ISSUE #12152: No information popover when autocomplete suggestions are showing
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

When hovering over a variable/method, a popover showing information about that variable/method normally appears. However, when there is an autocomplete popover, the information popover does not show up. 

https://github.com/zed-industries/zed/assets/50590465/b5f35ed7-5196-49a8-a153-4fec6f16f216

### Environment

Zed: v0.136.2 (Zed)
OS: macOS 14.5.0
Memory: 16 GiB
Architecture: aarch64

### If applicable, add mockups / screenshots to help explain present your vision of the feature

_No response_

### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue.

_No response_

### PR DESCRIPTION
**Pull request by effdotsh**
Allows information popovers to display while autocomplete suggestions are also being displayed.

Before: 

https://github.com/zed-industries/zed/assets/50590465/a61f0b4e-1521-42de-84fd-c5a3586b74ab

After:

https://github.com/zed-industries/zed/assets/50590465/ad2daae7-aebc-4c71-ad4c-f9d2deb6d0d0

Release Notes:
- Fixed #12152

### CODE REVIEW

File: crates/editor/src/element.rs [hunk 1]
```diff
@@ -4317,8 +4317,7 @@ impl Element for EditorElement {
                         &snapshot,
                         cx,
                     );
-
```
Comment (maxdeviant) [inline crates/editor/src/element.rs:4320]: Can we keep this blank line since it's not needed for the change?

**Review by SomeoneToIgnore — COMMENTED** [review]
Looks nice, but let's add a test for this so it does not degrade over time.

We do not have a good API for using the mouse, but what we can do, is to use a hover popover test as base:

https://github.com/zed-industries/zed/blob/f56558357893603d6a149e570062244425c6a4e3/crates/editor/src/hover_popover.rs#L633

but have two carets placed, then we can trigger a `Hover` action (as in any of the tests form the module linked above), and the completions, as in https://github.com/zed-industries/zed/blob/f56558357893603d6a149e570062244425c6a4e3/crates/editor/src/editor_tests.rs#L6519

**Comment by effdotsh** [comment]
@SomeoneToIgnore I wrote a test using the two you linked as a guide. It opens the autocomplete prompt, hovers for a popover, completes the autocomplete, then moves the mouse to dismiss the information popover. Do you think this is sufficient?

**Review by SomeoneToIgnore — APPROVED** [review]
Thank you, this scenario is even better, since checks this all for the same place.
I would have checked that the hover data is present, but we sort of do that implicitly by ensuring that we issue the hover LSP request after the completion.

**Pull request merged**